### ISSUE TITLE
💅  False-positive`useAdjacentOverloadSignatures` for static and none-static methods

### ISSUE THREAD

**Issue by owanturist** [biomejs/biome#8345]
### Environment information

```bash
CLI:
  Version:                      2.3.8
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v20.13.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/10.5.2

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               false
  VCS enabled:                  true

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  unset
  Enabled rules:
    a11y/noAccessKey
    a11y/noAriaHiddenOnFocusable
    a11y/noAriaUnsupportedElements
    a11y/noAutofocus
    a11y/noDistractingElements
    a11y/noHeaderScope
    a11y/noInteractiveElementToNoninteractiveRole
    a11y/noLabelWithoutControl
    a11y/noNoninteractiveElementInteractions
    a11y/noNoninteractiveElementToInteractiveRole
    a11y/noNoninteractiveTabindex
    a11y/noPositiveTabindex
    a11y/noRedundantAlt
    a11y/noRedundantRoles
    a11y/noStaticElementInteractions
    a11y/noSvgWithoutTitle
    a11y/useAltText
    a11y/useAnchorContent
    a11y/useAriaActivedescendantWithTabindex
    a11y/useAriaPropsForRole
    a11y/useAriaPropsSupportedByRole
    a11y/useButtonType
    a11y/useFocusableInteractive
    a11y/useGenericFontNames
    a11y/useHeadingContent
    a11y/useHtmlLang
    a11y/useIframeTitle
    a11y/useKeyWithClickEvents
    a11y/useKeyWithMouseEvents
    a11y/useMediaCaption
    a11y/useSemanticElements
    a11y/useValidAnchor
    a11y/useValidAriaProps
    a11y/useValidAriaRole
    a11y/useValidAriaValues
    a11y/useValidAutocomplete
    a11y/useValidLang
    complexity/noAdjacentSpacesInRegex
    complexity/noArguments
    complexity/noBannedTypes
    complexity/noCommaOperator
    complexity/noEmptyTypeParameters
    complexity/noExcessiveCognitiveComplexity
    complexity/noExcessiveLinesPerFunction
    complexity/noExcessiveNestedTestSuites
    complexity/noExtraBooleanCast
    complexity/noFlatMapIdentity
    complexity/noForEach
    complexity/noImplicitCoercions
    complexity/noImportantStyles
    complexity/noStaticOnlyClass
    complexity/noThisInStatic
    complexity/noUselessCatch
    complexity/noUselessConstructor
    complexity/noUselessContinue
    complexity/noUselessEmptyExport
    complexity/noUselessEscapeInRegex
    complexity/noUselessFragments
    complexity/noUselessLabel
    complexity/noUselessLoneBlockStatements
    complexity/noUselessRename
    complexity/noUselessStringConcat
    complexity/noUselessStringRaw
    complexity/noUselessSwitchCase
    complexity/noUselessTernary
    complexity/noUselessThisAlias
    complexity/noUselessTypeConstraint
    complexity/noUselessUndefinedInitialization
    complexity/noVoid
    complexity/useArrowFunction
    complexity/useDateNow
    complexity/useFlatMap
    complexity/useIndexOf
    complexity/useLiteralKeys
    complexity/useNumericLiterals
    complexity/useOptionalChain
    complexity/useRegexLiterals
    complexity/useSimpleNumberKeys
    complexity/useSimplifiedLogicExpression
    complexity/useWhile
    correctness/noChildrenProp
    correctness/noConstAssign
    correctness/noConstantCondition
    correctness/noConstantMathMinMaxClamp
    correctness/noConstructorReturn
    correctness/noEmptyCharacterClassInRegex
    correctness/noEmptyPattern
    correctness/noGlobalDirnameFilename
    correctness/noGlobalObjectCalls
    correctness/noInnerDeclarations
    correctness/noInvalidBuiltinInstantiation
    correctness/noInvalidConstructorSuper
    correctness/noInvalidDirectionInLinearGradient
    correctness/noInvalidGridAreas
    correctness/noInvalidPositionAtImportRule
    correctness/noInvalidUseBeforeDeclaration
    correctness/noMissingVarFunction
    correctness/noNestedComponentDefinitions
    correctness/noNodejsModules
    correctness/noNonoctalDecimalEscape
    correctness/noPrecisionLoss
    correctness/noPrivateImports
    correctness/noProcessGlobal
    correctness/noQwikUseVisibleTask
    correctness/noReactPropAssignments
    correctness/noRenderReturnValue
    correctness/noRestrictedElements
    correctness/noSelfAssign
    correctness/noSetterReturn
    correctness/noSolidDestructuredProps
    correctness/noStringCaseMismatch
    correctness/noSwitchDeclarations
    correctness/noUndeclaredDependencies
    correctness/noUndeclaredVariables
    correctness/noUnknownFunction
    correctness/noUnknownMediaFeatureName
    correctness/noUnknownProperty
    correctness/noUnknownPseudoClass
    correctness/noUnknownPseudoElement
    correctness/noUnknownTypeSelector
    correctness/noUnknownUnit
    correctness/noUnmatchableAnbSelector
    correctness/noUnreachable
    correctness/noUnreachableSuper
    correctness/noUnsafeFinally
    correctness/noUnsafeOptionalChaining
    correctness/noUnusedFunctionParameters
    correctness/noUnusedImports
    correctness/noUnusedLabels
    correctness/noUnusedPrivateClassMembers
    correctness/noUnusedVariables
    correctness/noVoidElementsWithChildren
    correctness/noVoidTypeReturn
    correctness/useExhaustiveDependencies
    correctness/useGraphqlNamedOperations
    correctness/useHookAtTopLevel
    correctness/useImageSize
    correctness/useImportExtensions
    correctness/useIsNan
    correctness/useJsonImportAttributes
    correctness/useJsxKeyInIterable
    correctness/useParseIntRadix
    correctness/useQwikClasslist
    correctness/useSingleJsDocAsterisk
    correctness/useUniqueElementIds
    correctness/useValidForDirection
    correctness/useValidTypeof
    correctness/useYield
    nursery/noContinue
    nursery/noDeprecatedImports
    nursery/noDuplicateDependencies
    nursery/noDuplicatedSpreadProps
    nursery/noEmptySource
    nursery/noEqualsToNull
    nursery/noFloatingPromises
    nursery/noForIn
    nursery/noImportCycles
    nursery/noIncrementDecrement
    nursery/noJsxLiterals
    nursery/noLeakedRender
    nursery/noMisusedPromises
    nursery/noMultiStr
    nursery/noNextAsyncClientComponent
    nursery/noParametersOnlyUsedInRecursion
    nursery/noProto
    nursery/noReactForwardRef
    nursery/noShadow
    nursery/noSyncScripts
    nursery/noTernary
    nursery/noUnknownAttribute
    nursery/noUnnecessaryConditions
    nursery/noUnresolvedImports
    nursery/noUnusedExpressions
    nursery/noUselessCatchBinding
    nursery/noUselessUndefined
    nursery/noVueDataObjectDeclaration
    nursery/noVueDuplicateKeys
    nursery/noVueReservedKeys
    nursery/noVueReservedProps
    nursery/noVueVIfWithVFor
    nursery/useArraySortCompare
    nursery/useConsistentArrowReturn
    nursery/useConsistentGraphqlDescriptions
    nursery/useDeprecatedDate
    nursery/useExhaustiveSwitchCases
    nursery/useExplicitType
    nursery/useFind
    nursery/useMaxParams
    nursery/useQwikMethodUsage
    nursery/useQwikValidLexicalScope
    nursery/useSortedClasses
    nursery/useSpread
    nursery/useUniqueGraphqlOperationName
    nursery/useVueDefineMacrosOrder
    nursery/useVueHyphenatedAttributes
    nursery/useVueMultiWordComponentNames
    nursery/useVueValidVBind
    nursery/useVueValidVElse
    nursery/useVueValidVElseIf
    nursery/useVueValidVHtml
    nursery/useVueValidVIf
    nursery/useVueValidVOn
    nursery/useVueValidVText
    performance/noAccumulatingSpread
    performance/noAwaitInLoops
    performance/noBarrelFile
    performance/noDelete
    performance/noDynamicNamespaceImportAccess
    performance/noImgElement
    performance/noNamespaceImport
    performance/noReExportAll
    performance/noUnwantedPolyfillio
    performance/useGoogleFontPreconnect
    performance/useSolidForComponent
    performance/useTopLevelRegex
    security/noBlankTarget
    security/noDangerouslySetInnerHtml
    security/noDangerouslySetInnerHtmlWithChildren
    security/noGlobalEval
    security/noSecrets
    style/noCommonJs
    style/noDefaultExport
    style/noDescendingSpecificity
    style/noDoneCallback
    style/noEnum
    style/noExportedImports
    style/noHeadElement
    style/noImplicitBoolean
    style/noInferrableTypes
    style/noMagicNumbers
    style/noNamespace
    style/noNegationElse
    style/noNestedTernary
    style/noNonNullAssertion
    style/noParameterAssign
    style/noParameterProperties
    style/noProcessEnv
    style/noRestrictedGlobals
    style/noRestrictedImports
    style/noRestrictedTypes
    style/noShoutyConstants
    style/noSubstr
    style/noUnusedTemplateLiteral
    style/noUselessElse
    style/noValueAtRule
    style/noYodaExpression
    style/useArrayLiterals
    style/useAsConstAssertion
    style/useAtIndex
    style/useBlockStatements
    style/useCollapsedElseIf
    style/useCollapsedIf
    style/useComponentExportOnlyModules
    style/useConsistentArrayType
    style/useConsistentBuiltinInstantiation
    style/useConsistentCurlyBraces
    style/useConsistentMemberAccessibility
    style/useConsistentObjectDefinitions
    style/useConsistentTypeDefinitions
    style/useConst
    style/useDefaultParameterLast
    style/useDefaultSwitchClause
    style/useDeprecatedReason
    style/useEnumInitializers
    style/useExplicitLengthCheck
    style/useExponentiationOperator
    style/useExportType
    style/useExportsLast
    style/useFilenamingConvention
    style/useForOf
    style/useFragmentSyntax
    style/useGraphqlNamingConvention
    style/useGroupedAccessorPairs
    style/useImportType
    style/useLiteralEnumMembers
    style/useNamingConvention
    style/useNodeAssertStrict
    style/useNodejsImportProtocol
    style/useNumberNamespace
    style/useNumericSeparators
    style/useObjectSpread
    style/useReactFunctionComponents
    style/useReadonlyClassProperties
    style/useSelfClosingElements
    style/useShorthandAssign
    style/useShorthandFunctionType
    style/useSingleVarDeclarator
    style/useSymbolDescription
    style/useTemplate
    style/useThrowNewError
    style/useThrowOnlyError
    style/useTrimStartEnd
    style/useUnifiedTypeSignatures
    suspicious/noAlert
    suspicious/noApproximativeNumericConstant
    suspicious/noArrayIndexKey
    suspicious/noAssignInExpressions
    suspicious/noAsyncPromiseExecutor
    suspicious/noBiomeFirstException
    suspicious/noBitwiseOperators
    suspicious/noCatchAssign
    suspicious/noClassAssign
    suspicious/noCommentText
    suspicious/noCompareNegZero
    suspicious/noConfusingLabels
    suspicious/noConfusingVoidType
    suspicious/noConsole
    suspicious/noConstEnum
    suspicious/noConstantBinaryExpressions
    suspicious/noControlCharactersInRegex
    suspicious/noDebugger
    suspicious/noDocumentCookie
    suspicious/noDocumentImportInPage
    suspicious/noDoubleEquals
    suspicious/noDuplicateAtImportRules
    suspicious/noDuplicateCase
    suspicious/noDuplicateClassMembers
    suspicious/noDuplicateCustomProperties
    suspicious/noDuplicateElseIf
    suspicious/noDuplicateFields
    suspicious/noDuplicateFontNames
    suspicious/noDuplicateJsxProps
    suspicious/noDuplicateObjectKeys
    suspicious/noDuplicateParameters
    suspicious/noDuplicateProperties
    suspicious/noDuplicateSelectorsKeyframeBlock
    suspicious/noDuplicateTestHooks
    suspicious/noEmptyBlock
    suspicious/noEmptyBlockStatements
    suspicious/noEmptyInterface
    suspicious/noEvolvingTypes
    suspicious/noExplicitAny
    suspicious/noExportsInTest
    suspicious/noExtraNonNullAssertion
    suspicious/noFallthroughSwitchClause
    suspicious/noFocusedTests
    suspicious/noFunctionAssign
    suspicious/noGlobalAssign
    suspicious/noGlobalIsFinite
    suspicious/noGlobalIsNan
    suspicious/noHeadImportInDocument
    suspicious/noImplicitAnyLet
    suspicious/noImportAssign
    suspicious/noImportantInKeyframe
    suspicious/noIrregularWhitespace
    suspicious/noLabelVar
    suspicious/noMisleadingCharacterClass
    suspicious/noMisleadingInstantiator
    suspicious/noMisplacedAssertion
    suspicious/noMisrefactoredShorthandAssign
    suspicious/noNonNullAssertedOptionalChain
    suspicious/noOctalEscape
    suspicious/noPrototypeBuiltins
    suspicious/noQuickfixBiome
    suspicious/noReactSpecificProps
    suspicious/noRedeclare
    suspicious/noRedundantUseStrict
    suspicious/noSelfCompare
    suspicious/noShadowRestrictedNames
    suspicious/noShorthandPropertyOverrides
    suspicious/noSkippedTests
    suspicious/noSparseArray
    suspicious/noSuspiciousSemicolonInJsx
    suspicious/noTemplateCurlyInString
    suspicious/noThenProperty
    suspicious/noTsIgnore
    suspicious/noUnassignedVariables
    suspicious/noUnknownAtRules
    suspicious/noUnsafeDeclarationMerging
    suspicious/noUnsafeNegation
    suspicious/noUselessEscapeInString
    suspicious/noUselessRegexBackrefs
    suspicious/noVar
    suspicious/noWith
    suspicious/useAdjacentOverloadSignatures
    suspicious/useAwait
    suspicious/useBiomeIgnoreFolder
    suspicious/useDefaultSwitchClauseLast
    suspicious/useErrorMessage
    suspicious/useGetterReturn
    suspicious/useGoogleFontDisplay
    suspicious/useGuardForIn
    suspicious/useIsArray
    suspicious/useIterableCallbackReturn
    suspicious/useNamespaceKeyword
    suspicious/useNumberToFixedDigitsArgument
    suspicious/useStaticResponseMethods
    suspicious/useStrictMode

Workspace:
  Open Documents:               0
```

### Rule name

useAdjacentOverloadSignatures

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=ZQB4AHAAbwByAHQAIABjAGwAYQBzAHMAIABLAGUAawAgAHsACgAgACAAcwB0AGEAdABpAGMAIABrAGUAawAoACkAOgBuAHUAbQBiAGUAcgAgAHsACgAgACAAIAAgAHIAZQB0AHUAcgBuACAAMAAKACAAIAB9AAoACgAgACAAYQBuAG8AdABoAGUAcgAoACkAOgAgAHMAdAByAGkAbgBnACAAewAKACAAIAAgACAAcgBlAHQAdQByAG4AIAAnACcACgAgACAAfQAKAAoAIAAgAGsAZQBrACgAKQA6AG4AdQBtAGIAZQByACAAewAKACAAIAAgACAAcgBlAHQAdQByAG4AIAAxAAoAIAAgAH0ACgB9AA%3D%3D

### Expected result

The rule should not report `static` and none-`static` methods named similarly but defined not consecutively after each other.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by tt-a1i** [comment]
I'd like to work on this. The fix should be straightforward - the grouping logic currently only uses method name as key, but should use (name, is_static) tuple to differentiate static methods from instance methods.
